kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java

```diff
@@ -76,9 +77,17 @@ public Class<? extends BaseConfig> getConfigType(String typeName) {
 
     @Override
     public T getInstance(String typeName, S context) {
-        if (typeNameToInstanceBuilder.containsKey(typeName)) {
-            InstanceBuilder<T, S> instanceBuilder = typeNameToInstanceBuilder.get(typeName).instanceBuilder();
-            return instanceBuilder == null ? null : instanceBuilder.construct(context);
+        if (typeNameToContributorDetails.containsKey(typeName)) {
+            final ContributionDetails<T, S> contributionDetails = typeNameToContributorDetails.get(typeName);
+            final boolean configurationRequired = contributionDetails.configurationDefinition().configurationRequired();
+            final boolean hasConfiguration = context.getConfig() != null;
+            if (!configurationRequired || hasConfiguration) {
+                InstanceBuilder<T, S> instanceBuilder = contributionDetails.instanceBuilder();
+                return instanceBuilder == null ? null : instanceBuilder.construct(context);
+            }
+            else {
+                throw new IllegalArgumentException("'" + typeName + "' requires configuration but none was supplied");
```

Comment:
I think throwing the exception here rather than returning null is acceptable as config validation is new and not breaking the backwards compatible contract. 

But given it is altering the contract of `getInstance` it could been seen the other way round. Equally we could just rip the plaster off and throw in all cases rather than returning `null`.

Reply:
> But given it is altering the contract of getInstance it could been seen the other way round. Equally we could just rip the plaster off and throw in all cases rather than returning null.

+1

Reply:
@k-wall I'm not clear which option you are `+1` to. I presume its the rip the plaster option but want to confirm before proceeding.

Reply:
+1 to throw in all cases and get more serious about the API stability after 0.3.0

Reply:
I agree, if we're breaking API compatibility _anyway_ let's use that as an opportunity to land quickly on what we now think is the right thing here.

Reply:
with that line of thinking should we go nuclear and instead of [deprecating](https://github.com/kroxylicious/kroxylicious/pull/612/files#diff-2b2472bcbf42d733f99e7fdefb53bfa81ed54e0dee510553aba0470337304000R68) `getConfigType` we should delete it?